File: frontend/src/app/domain/models/block.ts
export class Block {
  blockType: string;
  xStart: number;
  xEnd: number;
  yStart: number;
  yEnd: number;
  color: string;

  constructor(
    blockType: string,
    xStart: number,
    xEnd: number,
    yStart: number,
    yEnd: number,
    color: string
  ) {
    this.blockType = blockType;
    this.xStart = xStart;
    this.xEnd = xEnd;
    this.yStart = yStart;
    this.yEnd = yEnd;
    this.color = color;
  }
}

File: backend/src/main/java/com/joffrey/uberclone/businesslogic/models/Block.java
package com.joffrey.uberclone.businesslogic.models;

import com.joffrey.uberclone.businesslogic.usecases.map.BlockType;

public record Block(BlockType blockType, int xStart, int xEnd, int yStart, int yEnd, String color) {
    
}
